### DISCUSSION TITLE
How to use a single key binding to toggle hide and show panels and docks?

### DISCUSSION THREAD

**Discussion by Yevgnen** [zed-industries/zed#13342] [Feature Requests]
I Have the following key bindings 

- `ctrl-cmd-l`: `workspace::ToggleLeftDock`
- `ctrl-cmd-r`: `workspace::ToggleRightDock`

and 

- `ctrl-cmd-j`: `project_panel::ToggleFocus`
- `ctrl-cmd-a`: `assistant::ToggleFocus`
- `ctrl-cmd-o`: `outline_panel::ToggleFocus`

and hide all docks by default.

When I need to talk to the assistant I need to use `ctrl-cmd-a` open then assistant panel and `ctrl-cmd-r` to hide when finish.

When I do some file operation I need to use `ctrl-cmd-j` to open the project panel and then use `ctrl-cmd-l` to hide the panel back.

Is there a way to use single keybinding to hide and show the panels? For example,

- `ctrl-cmd-a` to open and hide the assistant panel.
- `ctrl-cmd-j` to open and hide the project panel.
- ...

Thanks!

**Comment by tepavcevic** [comment]
If I understood correctly, you need a way to close an open dock while it being focused?
Try something like this:
```
  {
    "context": "ProjectPanel && not_editing",
    "bindings": {
      "ctrl-cmd-j": "workspace::ToggleRightDock"
    }
  }
```
and the same goes for `"context": "AssistantPanel && not_editing"`.

**Reply by Yevgnen** [reply]
Thanks man! Works like a charm!
